### PR TITLE
Support for rds-postgres URL with rdsutils.BuildAuthToken

### PR DESCRIPTION
**Pull request by arthurzenika**
Related : #114

### CODE REVIEW

File: job.go [hunk 1]
```diff
@@ -221,6 +224,22 @@ func (j *Job) updateConnections() {
 				})
 				continue
 			}
+			if strings.HasPrefix(conn, "rds-postgres://") {
+				// reuse postgres SQLDriver by stripping rds- from connexion URL after building the RDS
+				// authentication token
+				conn = strings.TrimPrefix(conn, "rds-")
+				// FIXME - parsing twice the conn url to extract host & username
+				u, err := url.Parse(conn)
```
Comment (gaetanars) [inline job.go:232]:
```suggestion
				u, err := url.Parse(conn)
				if err != nil {
                                   panic(err)
                                 }
```

File: job.go [hunk 2]
```diff
@@ -221,6 +224,22 @@ func (j *Job) updateConnections() {
...
+				region := os.Getenv("AWS_REGION")
```
Comment (gaetanars) [inline job.go:233]: What appens if AWS_REGION env is not set ?

Comment (arthurzenika) [reply job.go:236]: Then the default region will be used, and the generated token will be wrong, the user will get a generic postgresql authentication error.

File: README.md [hunk 3]
```diff
@@ -212,6 +212,19 @@ environment.
 LOGLEVEL=info ./sql_exporter
 ```
 
+Database specific configurations
+--------------------------------
+
+For some database backends some special functionnality is available :
```
Comment (dewey) [inline README.md:218]: Small typo here in `functionality`

**Review by dewey — APPROVED** [review]
It sounds reasonable to me, but I don't have a way of testing it. If you tested it and it works for you I'm okay with merging it as it shouldn't be breaking anyone's workflow in any case if they don't use it.

**Review by dewey — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -212,6 +212,19 @@ environment.
...
+
+* cloudsql-postgres : a special `*` caracter can be used to query all databases
```
[the remark this answers is not in the document]
Comment (arthurzenika) [reply README.md:220]:
```suggestion
* cloudsql-postgres: a special `*` caracter can be used to query all databases
```

File: job.go [hunk 5]
```diff
@@ -221,6 +224,25 @@ func (j *Job) updateConnections() {
...
 				continue
 			}
+			if strings.HasPrefix(conn, "rds-postgres://") {
+				// reuse postgres SQLDriver by stripping rds- from connexion URL after building the RDS
+				// authentication token
+				conn = strings.TrimPrefix(conn, "rds-")
+				// FIXME - parsing twice the conn url to extract host & username
+				u, err := url.Parse(conn)
+				if err != nil {
+                                   panic(err)
```
Comment (arthurzenika) [inline job.go:234]:
```suggestion
  				   level.Error(j.log).Log("msg", "Failed to parse URL", "url", conn, "err", err)
				   continue
```

File: job.go [hunk 6]
```diff
@@ -221,6 +224,25 @@ func (j *Job) updateConnections() {
...
+                                 }
+				region := os.Getenv("AWS_REGION")
+				sess := session.Must(session.NewSessionWithOptions(session.Options{
+					SharedConfigState: session.SharedConfigEnable,
+				}))
+				token, err := rdsutils.BuildAuthToken(u.Host, region, u.User.Username(), sess.Config.Credentials)
+				if err != nil {
+					panic(err)
```
[the remark this answers is not in the document]
Comment (arthurzenika) [reply job.go:242]:
```suggestion
					level.Error(j.log).Log("msg", "Failed to parse URL", "url", conn, "err", err)
					continue
```

**Comment by arthurzenika** [comment]
Thanks for the review and the approval. We have a running version running in one environment so far.

**Pull request merged**